File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares provides functions for operating on the square of sums and sum of squares
package diffsquares

// SumOfSquares calculates the sum of squares up to a given number
func SumOfSquares(n int) int {
	total := 0
	for i := 0; i <= n; i++ {
		total += (i * i)
	}
	return total
}

// SquareOfSum calculates the square of sums up to a given number
func SquareOfSum(n int) int {
	total := 0
	for i := 0; i <= n; i++ {
		total += i
	}
	return total * total
}

// Difference calculates the difference between the square of sums and sum of squares
func Difference(n int) int {
	sumOfSquares := 0
	sum := 0
	for i := 0; i <= n; i++ {
		sum += i
		sumOfSquares += (i * i)
	}

	return (sum * sum) - sumOfSquares
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(private val n: Int) {
    private fun Int.squared() = this * this

    fun sumOfSquares(): Int {
        return (1..n).sumOf { it.squared() }
    }

    fun squareOfSum(): Int {
        return (1..n).sum().squared()
    }

    fun difference(): Int {
        return squareOfSum() - sumOfSquares()
    }
}
